### PR TITLE
Update regex for email address to be aligned on RFC 5322

### PR DESCRIPTION
**Pull request by yostyle**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other :

## Content

Update regex for email address to be aligned on RFC 5322
Similar to [Matrix React SDK](https://github.com/matrix-org/matrix-react-sdk/blob/c28f3164bda5de130faf465f2d819726c2d5690e/src/email.ts#L19)

## Motivation and context

Some characters in email address are not allowed in Element Android but should be accepted according to RFC 5322.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by yostyle** [comment]
Signed-off-by: yostyle <y.pintas@gmail.com>

File: vector/src/main/java/im/vector/app/core/extensions/BasicExtensions.kt [hunk 1]
```diff
@@ -29,7 +30,7 @@ inline fun <T> T.ooi(block: (T) -> Unit): T = also(block)
 /**
  * Check if a CharSequence is an email.
  */
-fun CharSequence.isEmail() = Patterns.EMAIL_ADDRESS.matcher(this).matches()
+fun CharSequence.isEmail() = this.isEmail()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/extensions/BasicExtensions.kt:33]: Reading it from GitHub, it looks like an infinite loop (it's maybe one?).
Maybe remove this extension for clarity. The call site can import `org.matrix.android.sdk.api.extensions.isEmail` instead?

Comment (yostyle) [reply vector/src/main/java/im/vector/app/core/extensions/BasicExtensions.kt]: Yes I removed this extension.

**Review by bmarty — APPROVED** [review]
Thanks!

**Pull request merged**